test/extended/security/scc.go

```diff
@@ -83,6 +86,60 @@ var _ = g.Describe("[sig-auth][Feature:SecurityContextConstraints] ", func() {
 			t.Fatalf("missing error: %v", err)
 		}
 	})
+
+	g.It("TestPodUpdateSCCEnforcement with service account", func() {
+		t := g.GinkgoT()
+
+		clusterAdminKubeClientset := oc.AdminKubeClient()
+
+		projectName := oc.Namespace()
+
+		sa := createServiceAccount(ctx, oc, projectName)
+		createPodAdminRoleOrDie(ctx, oc, sa)
+		restrictedClient, _ := createClientFromServiceAccount(oc, sa)
+
+		// so cluster-admin can create privileged pods, but harold cannot.  This means that harold should not be able
+		// to update the privileged pods either, even if he lies about its privileged nature
+		privilegedPod := getPrivilegedPod("unsafe")
+
+		if _, err := restrictedClient.CoreV1().Pods(projectName).Create(ctx, privilegedPod, metav1.CreateOptions{}); !isForbiddenBySCC(err) {
+			t.Fatalf("missing forbidden: %v", err)
+		}
+
+		actualPod, err := clusterAdminKubeClientset.CoreV1().Pods(projectName).Create(ctx, privilegedPod, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		actualPod.Spec.Containers[0].Image = "something-nefarious"
+		if _, err := restrictedClient.CoreV1().Pods(projectName).Update(ctx, actualPod, metav1.UpdateOptions{}); !isForbiddenBySCC(err) {
+			t.Fatalf("missing forbidden: %v", err)
+		}
+
+		// try to connect to /exec subresource as harold
+		restrictedClientCorev1Rest := restrictedClient.CoreV1().RESTClient()
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		result := &metav1.Status{}
+		err = restrictedClientCorev1Rest.Post().
+			Resource("pods").
+			Namespace(projectName).
+			Name(actualPod.Name).
+			SubResource("exec").
+			Param("container", "first").
+			Do(ctx).
+			Into(result)
+		if !isForbiddenBySCCExecRestrictions(err) {
+			t.Fatalf("missing forbidden by SCCExecRestrictions: %v", err)
+		}
+
+		// try to lie about the privileged nature
+		actualPod.Spec.HostPID = false
+		if _, err := restrictedClient.CoreV1().Pods(projectName).Update(context.Background(), actualPod, metav1.UpdateOptions{}); err == nil {
+			t.Fatalf("missing error: %v", err)
+		}
+	})
```

Comment:
This looks like a copy-pasta of the above. Could you just derive the variables (I believe it's just the restConfig for the user?) from the code and turn this into a function that's run from the two tests?